### PR TITLE
rename service port to disable istio manual protocol sniffing

### PR DESCRIPTION
**Pull request by ravisingal**

### CODE REVIEW

**Review by mohit-a21 — APPROVED** [review]

**Comment by ravisingal** [comment]
istio-proxy logs after changing the port name in entity-service:
```
2020-10-01T05:59:18.658690Z     debug   envoy http      [external/envoy/source/common/http/conn_manager_impl.cc:268] [C208765] new stream
2020-10-01T05:59:18.658889Z     debug   envoy http      [external/envoy/source/common/http/conn_manager_impl.cc:782] [C208765][S1782578881682651723] request headers complete (end_stream=false):
':authority', 'entity-service:50061'
':path', '/org.hypertrace.entity.data.service.v1.EntityDataService/getByTypeAndIdentifyingProperties'
':method', 'POST'
':scheme', 'https'
'content-type', 'application/grpc'
'te', 'trailers'
'user-agent', 'grpc-java-netty/1.31.1'
'grpc-accept-encoding', 'gzip'
'x-tenant-id', '3e761879-c77b-4d8f-a075-62ff28e8fa8a'
'x-forwarded-proto', 'http'
'x-request-id', '8632a8bb-d76c-92f8-bf04-ec08a44cca72'
'x-envoy-decorator-operation', 'entity-service.traceable.svc.cluster.local:50061/*'
'x-envoy-peer-metadata', 'ChoKCkNMVVNURVJfSUQSDBoKS3ViZXJuZXRlcwocCgxJTlNUQU5DRV9JUFMSDBoKMTAuNDguNS41MgrOAwoGTEFCRUxTEsMDKsADChcKA2FwcBIQGg50cmFjZS1lbnJpY2hlcgo3Chhjb250cm9sbGVyLXJldmlzaW9uLWhhc2gSGxoZdHJhY2UtZW5yaWNoZXItNzZmN2I1YzQ1NAoZC
gxpc3Rpby5pby9yZXYSCRoHZGVmYXVsdAogCgxwcm9kdWN0Z3JvdXASEBoOdHJhY2UtcGlwZWxpbmUKGwoHcmVsZWFzZRIQGg50cmFjZS1lbnJpY2hlcgokChlzZWN1cml0eS5pc3Rpby5pby90bHNNb2RlEgcaBWlzdGlvCjMKH3NlcnZpY2UuaXN0aW8uaW8vY2Fub25pY2FsLW5hbWUSEBoOdHJhY2UtZW5yaWNoZXIK
Lwojc2VydmljZS5pc3Rpby5pby9jYW5vbmljYWwtcmV2aXNpb24SCBoGbGF0ZXN0CjgKInN0YXRlZnVsc2V0Lmt1YmVybmV0ZXMuaW8vcG9kLW5hbWUSEhoQdHJhY2UtZW5yaWNoZXItMAoiChV0cmFjZWFibGVhaS1pbmplY3Rpb24SCRoHZW5hYmxlZAooCht0cmFjZWFibGVhaS1pbnN0cnVtZW50YXRpb24SCRoHZW5
hYmxlZAoaCgdNRVNIX0lEEg8aDWNsdXN0ZXIubG9jYWwKGgoETkFNRRISGhB0cmFjZS1lbnJpY2hlci0wChgKCU5BTUVTUEFDRRILGgl0cmFjZWFibGUKVQoFT1dORVISTBpKa3ViZXJuZXRlczovL2FwaXMvYXBwcy92MS9uYW1lc3BhY2VzL3RyYWNlYWJsZS9zdGF0ZWZ1bHNldHMvdHJhY2UtZW5yaWNoZXIKzwMKEV
BMQVRGT1JNX01FVEFEQVRBErkDKrYDCoUBChtnY3BfZ2NlX2luc3RhbmNlX2NyZWF0ZWRfYnkSZhpkcHJvamVjdHMvMzQ1MjUwODM0OTg2L3pvbmVzL3VzLXdlc3QxLWIvaW5zdGFuY2VHcm91cE1hbmFnZXJzL2drZS1zYWFzLWRhdmUtZGVmYXVsdC1wb29sLTBhNzc0ODRhLWdycAosChNnY3BfZ2NlX2luc3RhbmNlX
2lkEhUaEzE4NTgwNzg0ODAyMDY4MTQxNDcKcQoZZ2NwX2djZV9pbnN0YW5jZV90ZW1wbGF0ZRJUGlJwcm9qZWN0cy8zNDUyNTA4MzQ5ODYvZ2xvYmFsL2luc3RhbmNlVGVtcGxhdGVzL2drZS1zYWFzLWRhdmUtZGVmYXVsdC1wb29sLWE5MDQ2MWMzCiMKFGdjcF9na2VfY2x1c3Rlcl9uYW1lEgsaCXNhYXMtZGF2ZQoc
CgxnY3BfbG9jYXRpb24SDBoKdXMtd2VzdDEtYgoiCgtnY3BfcHJvamVjdBITGhFkYXRhc2VydmljZXMtMDAwMQokChJnY3BfcHJvamVjdF9udW1iZXISDhoMMzQ1MjUwODM0OTg2ChwKD1NFUlZJQ0VfQUNDT1VOVBIJGgdkZWZhdWx0CiEKDVdPUktMT0FEX05BTUUSEBoOdHJhY2UtZW5yaWNoZXI='
'x-envoy-peer-metadata-id', 'sidecar~10.48.5.52~trace-enricher-0.traceable~traceable.svc.cluster.local'
'x-envoy-attempt-count', '1'
'traceparent', '00-81a6dcc409c9181bf616f91e9881652d-b1b198da2f176f97-01'
'grpc-trace-bin', 'AACBptzECckYG/YW+R6YgWUtAbGxmNovF2+XAgE'
'x-b3-traceid', '81a6dcc409c9181bf616f91e9881652d'
'x-b3-spanid', 'b1b198da2f176f97'
'x-b3-sampled', '1'
2020-10-01T05:59:18.659161Z     debug   envoy filter    [external/istio_proxy/src/envoy/http/authn/http_filter.cc:55] AuthenticationFilter::decodeHeaders with config
policy {
  peers {
    mtls {
    }
  }
}
2020-10-01T05:59:18.659189Z     debug   envoy filter    [external/istio_proxy/src/envoy/http/authn/authenticator_base.cc:98] [C208765] validateX509 mode STRICT: ssl=true, has_user=true
2020-10-01T05:59:18.659195Z     debug   envoy filter    [external/istio_proxy/src/envoy/http/authn/authenticator_base.cc:77] [C208765] trust domain validation succeeded
2020-10-01T05:59:18.659200Z     debug   envoy filter    [external/istio_proxy/src/envoy/http/authn/filter_context.cc:35] Set peer from X509: cluster.local/ns/traceable/sa/default
2020-10-01T05:59:18.659206Z     debug   envoy filter    [external/istio_proxy/src/envoy/http/authn/filter_context.cc:61] Set principal from peer: cluster.local/ns/traceable/sa/default
2020-10-01T05:59:18.659210Z     debug   envoy filter    [external/istio_proxy/src/envoy/http/authn/origin_authenticator.cc:106] Origin authenticator succeeded
2020-10-01T05:59:18.659291Z     debug   envoy filter    [external/istio_proxy/src/envoy/http/authn/http_filter.cc:88] Saved Dynamic Metadata:
fields {
  key: "request.auth.principal"
  value {
    string_value: "cluster.local/ns/traceable/sa/default"
  }
}
fields {
  key: "source.namespace"
  value {
    string_value: "traceable"
  }
}
fields {
  key: "source.principal"
  value {
    string_value: "cluster.local/ns/traceable/sa/default"
  }
}
fields {
  key: "source.user"
  value {
    string_value: "cluster.local/ns/traceable/sa/default"
  }
}
2020-10-01T05:59:18.659333Z     debug   envoy router    [external/envoy/source/common/router/router.cc:477] [C208765][S1782578881682651723] cluster 'inbound|50061|entity-service|entity-service.traceable.svc.cluster.local' match for URL '/org.hypertrace.entity.data.service.v1.EntityDataService/getByTypeAndIdentifyingProperties'
2020-10-01T05:59:18.659401Z     debug   envoy router    [external/envoy/source/common/router/router.cc:634] [C208765][S1782578881682651723] router decoding headers:
':authority', 'entity-service:50061'
':path', '/org.hypertrace.entity.data.service.v1.EntityDataService/getByTypeAndIdentifyingProperties'
':method', 'POST'
':scheme', 'http'
'content-type', 'application/grpc'
'te', 'trailers'
'user-agent', 'grpc-java-netty/1.31.1'
'grpc-accept-encoding', 'gzip'
'x-tenant-id', '3e761879-c77b-4d8f-a075-62ff28e8fa8a'
'x-forwarded-proto', 'http'
'x-request-id', '8632a8bb-d76c-92f8-bf04-ec08a44cca72'
'x-envoy-attempt-count', '1'
'x-forwarded-client-cert', 'By=spiffe://cluster.local/ns/traceable/sa/default;Hash=62b781f2264117bfd21636c7dcd242da3c64b8502edfd95b3d96ff3ff37cac39;Subject="";URI=spiffe://cluster.local/ns/traceable/sa/default'
'traceparent', '00-81a6dcc409c9181bf616f91e9881652d-5ce1c9b77e948ddd-01'
'grpc-trace-bin', 'AACBptzECckYG/YW+R6YgWUtAVzhybd+lI3dAgE'
'x-b3-traceid', '81a6dcc409c9181bf616f91e9881652d'
'x-b3-spanid', '5ce1c9b77e948ddd'
'x-b3-sampled', '1'
2020-10-01T05:59:18.659429Z     debug   envoy pool      [external/envoy/source/common/http/conn_pool_base.cc:118] [C159037] using existing connection
2020-10-01T05:59:18.659438Z     debug   envoy pool      [external/envoy/source/common/http/conn_pool_base.cc:68] [C159037] creating stream
2020-10-01T05:59:18.659455Z     debug   envoy router    [external/envoy/source/common/router/upstream_request.cc:317] [C208765][S1782578881682651723] pool ready
2020-10-01T05:59:18.659565Z     debug   envoy http      [external/envoy/source/common/http/conn_manager_impl.cc:1337] [C208765][S1782578881682651723] request end stream
```

**Review by kotharironak — APPROVED** [review]

File: helm/templates/service.yaml [hunk 1]
```diff
@@ -10,6 +10,6 @@ spec:
     - port: {{ .Values.service.port }}
       targetPort: grpc-port
       protocol: TCP
-      name: grpc-port
+      name: {{ .Chart.Name }}
```
Comment (tim-mwangi) [inline helm/templates/service.yaml:13]: why are we doing this @ravisingal ?

Comment (davexroth) [reply helm/templates/service.yaml:13]: same question :)

Comment (ravisingal) [reply helm/templates/service.yaml:13]: istio can automatically detect http and http/2 traffic. with manual protocol detection, client was complaining about protocol mismatch between grpc server and client (server using ssl and client was using plaintext).

Comment (davexroth) [reply helm/templates/service.yaml:13]: I thought the issue you were having with mTLS between some services was that the protocol sniffing was causing issues, so we were going to go back and use manual protocol specification.

Can you explain the protocol mismatch a little more.   The client and server are both grpc right?  Why is there a mismatch between ssl and plaintext?

**Pull request merged**